### PR TITLE
Move addwindow and rcdocref to a provider

### PR DESCRIPTION
**Pull request by ylvaselling**

### CODE REVIEW

File: src/windowmanagement/WindowLayout/WindowLayout.tsx [hunk 1]
```diff
@@ -10,6 +10,8 @@ import { menuItemsDB } from '../data/MenuItems';
 import { TaskBar } from '../TaskBar';
 import { TopMenuBar } from '../TopMenuBar';
 
+import { useWindowManagerProvider } from './WindowLayoutProvider';
```
[the remark this answers is not in the document]
Comment (engbergandreas) [reply src/windowmanagement/WindowLayout/WindowLayout.tsx:13]: I'm the opposite I like that we make it clear that the provider is for the layout

File: src/windowmanagement/WindowLayout/WindowLayoutProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+export const WindowManagerContext = createContext<ProviderProps | null>(null);
+
+export function WindowManagerProvider({ children }: { children: React.ReactNode }) {
+  const rcDocRef = useRef<DockLayout>(null);
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  function isPanelDataInstance(obj: any) {
+    return obj && Array.isArray(obj.tabs);
```
Comment (engbergandreas) [inline src/windowmanagement/WindowLayout/WindowLayoutProvider.tsx:19]: Not sure if we tried this the first time, but shouldn't we be able to do `obj: object`?

**Comment by engbergandreas** [comment]
don't forget to run lint

**Review by engbergandreas — APPROVED** [review]

File: src/windowmanagement/WindowLayout/WindowLayout.tsx [hunk 3]
```diff
@@ -10,6 +10,8 @@ import { menuItemsDB } from '../data/MenuItems';
 import { TaskBar } from '../TaskBar';
```
Comment (WeirdRubberDuck) [inline src/windowmanagement/WindowLayout/WindowLayout.tsx:10]: Note that there are a bunch of imports that can be removed in this file now (on the first three lines)

**Comment by WeirdRubberDuck** [comment]
Can we maybe merge this? :) @ylvaselling

File: src/windowmanagement/WindowLayout/WindowLayoutProvider.tsx (hunk 2 above)
Comment (ylvaselling) [reply src/windowmanagement/WindowLayout/WindowLayoutProvider.tsx:19]: Fixed it

**Pull request merged**